### PR TITLE
feat: add new template for Bot to response base on Ces Wiki

### PR DESCRIPTION
**Pull request by ces-phatnguyen**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/Code-Engine-Studio/ces-semantic-release/blob/main/README.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #CU-86cu176fk

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by ces-sonnguyen** [comment]
Please sync with master branch.

**Review by ces-vinhhuynh — APPROVED** [review]

File: server/service.go [hunk 1]
```diff
@@ -15,15 +15,18 @@ const (
 	defaultSpellcheckLanguage = "English"
 )
 
-func (p *Plugin) processUserRequestToBot(context ai.ConversationContext) error {
+func (p *Plugin) processUserRequestToBot(context ai.ConversationContext, post *model.Post) error {
```
[the remark this answers is not in the document]
Comment (ces-phatnguyen) [reply server/service.go]: Updated

File: server/ai/prompts.go [hunk 2]
```diff
@@ -13,6 +13,10 @@ type Prompts struct {
 	getBuiltInTools func() []Tool
 }
 
+type ExtraPromptInformation struct {
+	Wiki string
+}
+
```
[the remark this answers is not in the document]
Comment (ces-phatnguyen) [reply server/ai/prompts.go]: I removed it

File: README.md [hunk 3]
```diff
@@ -33,6 +33,12 @@ This is forked project from the [Original Mattermost AI Plugin](https://github.c
 5. npm run dev
 6. Output files should be located in scripts/out
 
+If you want to regenerate the file `finalOutput.json` after manually editing the output files locally.
+
+npm run gen
+
+This will update the file `finalOutput.json` in scripts/out/final-output
+
```
Comment (ces-sonnguyen) [inline README.md:41]: Remove pls

File: server/conversation_context.go [hunk 4]
```diff
@@ -1,12 +1,16 @@
...
+	"fmt"
+
 	"github.com/mattermost/mattermost-plugin-ai/server/ai"
 	"github.com/mattermost/mattermost/server/public/model"
 )
 
 func (p *Plugin) MakeConversationContext(user *model.User, channel *model.Channel, post *model.Post) ai.ConversationContext {
 	context := ai.NewConversationContext(user, channel, post)
+	fmt.Printf("cesai-post %+v\n", post)
+	fmt.Printf("cesai-p%+v\n", p)
```
Comment (ces-sonnguyen) [inline server/conversation_context.go:13]: Remove log

File: server/conversation_context.go [hunk 5]
```diff
@@ -1,12 +1,16 @@
 package main
 
 import (
+	"fmt"
+
```
Comment (ces-sonnguyen) [inline server/conversation_context.go:5]: Format file

File: server/service.go [hunk 6]
```diff
@@ -109,6 +112,9 @@ func (p *Plugin) continueConversation(context ai.ConversationContext) error {
 			return err
 		}
 	} else {
+		// TODO: Fetch data from wiki here
+
+		context.Wiki += ""
```
Comment (ces-sonnguyen) [inline server/service.go:117]: Remove

File: server/service.go [hunk 7]
```diff
@@ -140,6 +146,8 @@ func (p *Plugin) continueThreadConversation(questionThreadData *ThreadData, orig
 	originalThread := formatThread(originalThreadData)
 
 	context.PromptParameters = map[string]string{"Thread": originalThread}
+	// TODO: Fetch data from wiki here
+
```
Comment (ces-sonnguyen) [inline server/service.go:150]: Remove

File: server/service.go [hunk 8]
```diff
@@ -24,6 +24,9 @@ func (p *Plugin) processUserRequestToBot(context ai.ConversationContext) error {
 }
 
 func (p *Plugin) newConversation(context ai.ConversationContext) error {
+	// TODO: Fetch data from wiki here
+
+	context.Wiki += ""
```
Comment (ces-sonnguyen) [inline server/service.go:29]: Remove

File: server/service.go (hunk 6 above)
Comment (ces-phatnguyen) [reply server/service.go]: removed

File: server/service.go (hunk 7 above)
Comment (ces-phatnguyen) [reply server/service.go]: removed

File: server/conversation_context.go (hunk 5 above)
Comment (ces-phatnguyen) [reply server/conversation_context.go]: updated

File: server/plugin.go [hunk 9]
```diff
@@ -31,6 +31,11 @@ const (
 //go:embed ai/prompts
 var promptsFolder embed.FS
 
+type QdrantClientType struct {
+	QdrantClient           any
+	QdrantCollectionClient any
+}
+
```
Comment (ces-sonnguyen) [inline server/plugin.go:38]: Remove

File: server/plugin.go [hunk 10]
```diff
@@ -55,6 +60,8 @@ type Plugin struct {
 
 	streamingContexts      map[string]context.CancelFunc
 	streamingContextsMutex sync.Mutex
+
+	qdrantClient QdrantClientType
```
Comment (ces-sonnguyen) [inline server/plugin.go:64]: import pb "github.com/qdrant/go-client/qdrant"

----

qdrantClient pb.QdrantClient

File: server/qdrant.go [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+	"context"
+	"fmt"
+	"time"
+
+	pb "github.com/qdrant/go-client/qdrant"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+func connectDb() (QdrantClientType, error) {
```
Comment (ces-sonnguyen) [inline server/qdrant.go:13]: func connectDb() (pb.QdrantClient, error) {

File: server/qdrant.go [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+	if err != nil {
+		fmt.Printf("fatal Could not get collections: %v", err)
+	} else {
+		fmt.Printf("List of collections: %s", r.GetCollections())
+	}
+	result := QdrantClientType{
+		QdrantClient:           qdrantClient,
+		QdrantCollectionClient: collections_client,
+	}
+	return result, nil
```
Comment (ces-sonnguyen) [inline server/qdrant.go:43]: return qdrantClient, nil

File: server/service.go [hunk 13]
```diff
@@ -140,6 +140,7 @@ func (p *Plugin) continueThreadConversation(questionThreadData *ThreadData, orig
 	originalThread := formatThread(originalThreadData)
 
 	context.PromptParameters = map[string]string{"Thread": originalThread}
+
```
Comment (ces-sonnguyen) [inline server/service.go:143]: Remove

**Review by ces-sonnguyen — APPROVED** [review]

**Pull request merged**